app/jobs/send_webmention_job.rb

```diff
@@ -0,0 +1,86 @@
+class SendWebmentionJob < ApplicationJob
+  queue_as :default
+
+  def endpoint_from_body(html)
+    doc = Nokogiri::HTML(html)
+  
+    if !doc.css('[rel~="webmention"]').css("[href]").empty?
+      doc.css('[rel~="webmention"]').css("[href]").attribute("href").value
+    elsif !doc.css('[rel="http://webmention.org/"]').css("[href]").empty?
+      doc.css('[rel="http://webmention.org/"]').css("[href]").attribute("href").value
+    elsif !doc.css('[rel="http://webmention.org"]').css("[href]").empty?
+      doc.css('[rel="http://webmention.org"]').css("[href]").attribute("href").value
+    end
+  end
+  
+  def endpoint_from_headers(header)
+    return unless header
+  
+    if (matches = header.match(/<([^>]+)>; rel="[^"]*\s?webmention\s?[^"]*"/))
+      matches[1]
+    elsif (matches = header.match(/<([^>]+)>; rel=webmention/))
+      matches[1]
+    elsif (matches = header.match(/rel="[^"]*\s?webmention\s?[^"]*"; <([^>]+)>/))
+      matches[1]
+    elsif (matches = header.match(/rel=webmention; <([^>]+)>/))
+      matches[1]
+    elsif (matches = header.match(/<([^>]+)>; rel="http:\/\/webmention\.org\/?"/))
+      matches[1]
+    elsif (matches = header.match(/rel="http:\/\/webmention\.org\/?"; <([^>]+)>/))
+      matches[1]
+    end
+  end
+  
+  # Some pages could return a relative link as their webmention endpoint.
+  # We need to translate this relative likn to an absolute one.
+  def uri_to_absolute(uri, req_uri)
+    abs_uri = URI.parse(uri)
+    if abs_uri.host
+      # Already absolute.
+      uri
+    else
+      abs_uri.host = req_uri.host
+      abs_uri.scheme = req_uri.scheme
+      abs_uri.port = req_uri.port
+      abs_uri
+    end
+  end
+  
+  def send_webmention(source, target, endpoint)
+    sp = Sponge.new
+    sp.timeout = 10
+    # Don't check SSL certificate here for backward compatibility, security risk
+    # is minimal.
+    sp.ssl_verify = false
+    sp.fetch(endpoint.to_s, :post, {
+      "source" => URI.encode_www_form_component(source),
+      "target" => URI.encode_www_form_component(target)
+    }, nil, {}, 3)
+  end
+
+  def perform(story)
```

Comment:
first big change: this now runs on a single story, so there's no longer any need to use `LAST_STORY_KEY`, iterate over multiple stories, etc.

Reply:
> If the job tries once and stops, great, that's a bit of implicit logic we don't have to maintain.

great, then we should be good to go! unless I'm wildly off the mark, SolidQueue doesn't do retries by default (need to slap a `retry_on` in the ActiveJob if you want that). so this should indeed try it once and stop.

Reply:
Great!